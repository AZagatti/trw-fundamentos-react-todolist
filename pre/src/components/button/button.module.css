.button {
  padding: 8px 16px;
  border-radius: 4px;
  border: none;
  cursor: pointer;
}
.base {
  background-color: #3f606b;
  color: #fff;
}
.edit {
  background-color: #1741a4;
  color: #fff;
  margin-right: 8px;
}
.confirm {
  background-color: #3eb726;
  color: #fff;
}
.remove {
  background-color: #c32121;
  color: #fff;
}
