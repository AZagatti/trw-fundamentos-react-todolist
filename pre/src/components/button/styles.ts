import styled, { css } from 'styled-components'

interface ButtonProps {
  $theme?: 'base' | 'edit' | 'confirm' | 'remove'
}

export const getTheme = ({ $theme }: ButtonProps) => {
  if ($theme === 'edit') {
    return css`
      background-color: #1741a4;
      color: #fff;
      margin-right: 8px;
    `
  }
  if ($theme === 'confirm') {
    return css`
      background-color: #3eb726;
      color: #fff;
    `
  }
  if ($theme === 'remove') {
    return css`
      background-color: #c32121;
      color: #fff;
    `
  }
  return css`
    background-color: #3f606b;
    color: #fff;
  `
}

export const Button = styled.button<ButtonProps>`
  padding: 8px 16px;
  border-radius: 4px;
  border: none;
  cursor: pointer;
  ${getTheme}
  &:disabled {
    background-color: #999;
    cursor: not-allowed;
  }
`
