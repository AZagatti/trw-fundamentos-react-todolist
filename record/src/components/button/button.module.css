button:disabled {
  background-color: #b1b1b1;
  cursor: not-allowed;
}
.button {
  padding: 8px 16px;
  border-radius: 4px;
  border: none;
  cursor: pointer;
  color: #fff;
}
.base {
  background-color: #3f606b;
}
.edit {
  background-color: #1741a4;
  margin-right: 8px;
}
.confirm {
  background-color: #3eb726;
}
.remove {
  background-color: #c32121;
}