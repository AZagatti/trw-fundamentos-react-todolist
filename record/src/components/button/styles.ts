import styled, { css } from 'styled-components'

interface ButtonProps {
  $theme?: 'base' | 'edit' | 'confirm' | 'remove'
}

const getTheme = ({ $theme }: ButtonProps) => {
  if ($theme === 'edit') {
    return css`
      background-color: #1741a4;
      margin-right: 8px;
    `
  }
  if ($theme === 'confirm') {
    return css`
      background-color: #3eb726;
    `
  }
  if ($theme === 'remove') {
    return css`
      background-color: #c32121;
    `
  }
  return css`
    background-color: #3f606b;
  `
}

export const Button = styled.button<ButtonProps>`
  padding: 8px 16px;
  border-radius: 4px;
  border: none;
  cursor: pointer;
  color: #fff;
  &:disabled {
    background-color: #b1b1b1;
    cursor: not-allowed;
  }
  ${getTheme}
`
