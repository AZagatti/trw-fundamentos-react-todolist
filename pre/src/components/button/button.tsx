import { ButtonHTMLAttributes, PropsWithChildren } from "react"
import styles from './button.module.css'

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  theme?: 'base' | 'edit' | 'confirm' | 'remove'
}

export const Button = ({ children, theme = 'base', className, ...props }: PropsWithChildren<ButtonProps>) => {
  return (
    <button
      className={`${className ?? ''} ${styles.button} ${styles[theme]}`}
      type="button"
      {...props}
    >
      {children}
    </button>
  )
}