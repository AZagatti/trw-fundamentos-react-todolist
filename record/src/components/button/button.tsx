import { ButtonHTMLAttributes, PropsWithChildren } from "react"
import * as S from './styles'

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  theme?: 'base' | 'edit' | 'confirm' | 'remove'
}

export const Button = ({
  children,
  theme = 'base',
  ...props
}: PropsWithChildren<ButtonProps>) => {
  return (
    <S.Button
      type="button"
      $theme={theme}
      {...props}
    >
      {children}
    </S.Button>
  )
}