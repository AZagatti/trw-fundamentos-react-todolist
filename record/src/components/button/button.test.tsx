import { render, screen } from '@testing-library/react'
import { Button } from './button'

describe('<Button />', () => {
  it.each(['base', 'edit', 'confirm', 'remove'] as const)(
    'can render with %s class',
    (theme) => {
      render(<Button theme={theme}>Button</Button>)
      expect(screen.getByText(/button/i)).toMatchSnapshot()
    }
  )
})
