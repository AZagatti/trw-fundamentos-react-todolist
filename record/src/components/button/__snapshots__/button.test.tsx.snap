// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > can render with base class 1`] = `
.c0 {
  padding: 8px 16px;
  border-radius: 4px;
  border: none;
  cursor: pointer;
  color: #fff;
  background-color: #3f606b;
}

.c0:disabled {
  background-color: #b1b1b1;
  cursor: not-allowed;
}

<button
  class="c0"
  type="button"
>
  Button
</button>
`;

exports[`<Button /> > can render with confirm class 1`] = `
.c0 {
  padding: 8px 16px;
  border-radius: 4px;
  border: none;
  cursor: pointer;
  color: #fff;
  background-color: #3eb726;
}

.c0:disabled {
  background-color: #b1b1b1;
  cursor: not-allowed;
}

<button
  class="c0"
  type="button"
>
  Button
</button>
`;

exports[`<Button /> > can render with edit class 1`] = `
.c0 {
  padding: 8px 16px;
  border-radius: 4px;
  border: none;
  cursor: pointer;
  color: #fff;
  background-color: #1741a4;
  margin-right: 8px;
}

.c0:disabled {
  background-color: #b1b1b1;
  cursor: not-allowed;
}

<button
  class="c0"
  type="button"
>
  Button
</button>
`;

exports[`<Button /> > can render with remove class 1`] = `
.c0 {
  padding: 8px 16px;
  border-radius: 4px;
  border: none;
  cursor: pointer;
  color: #fff;
  background-color: #c32121;
}

.c0:disabled {
  background-color: #b1b1b1;
  cursor: not-allowed;
}

<button
  class="c0"
  type="button"
>
  Button
</button>
`;
